vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -430,9 +429,9 @@ private List getInitContainerList(config){
     def initContainerSpecList = []
     if (config.initContainerImage && config.containerMountPath) {
         // regex [\W_] matches any non-word character equivalent to [^a-zA-Z0-9_]
-        def initContainerName = config.initContainerImage.replaceAll(/[\W_]/,"_" )
+        def initContainerName = config.initContainerImage.replaceAll(/[\W_]/,"-" )
```

Comment:
I strongly suggest chaining the toLowerCase method here, instead of lowering the case later.